### PR TITLE
UI themes (light, dark and black modes)

### PR DESCRIPTION
**Pull request by bjarniivarsson**
This adds light, dark and black mode themes, as well as support for automatic theme selection on windows 10/11 based on os settings. 
Dark theme:
image

Black theme:
image

Light theme:
image

### CODE REVIEW

File: Aga.Controls/Tree/NodeControls/BaseTextControl.cs [hunk 1]
```diff
@@ -219,23 +219,23 @@ private static Brush GetFrush(Color color)
 
 		private void CreateBrushes(TreeNodeAdv node, DrawContext context, string text, out Brush backgroundBrush, out Color textColor, out Font font, ref string label)
 		{
-			textColor = SystemColors.ControlText;
+            textColor = Parent.ForeColor;
```
Comment (PhyxionNL) [inline Aga.Controls/Tree/NodeControls/BaseTextControl.cs:222]: The formatting in this entire file is off.

File: Aga.Controls/Tree/NodeControls/NodeCheckBox.cs [hunk 2]
```diff
@@ -55,7 +55,14 @@ public override void Draw(TreeNodeAdv node, DrawContext context)
...
 			CheckState state = GetCheckState(node);
-			if (Application.RenderWithVisualStyles)
+
+            if (TreeViewAdv.CustomCheckRenderFunc != null)
+            {
+                TreeViewAdv.CustomCheckRenderFunc(context.Graphics, bounds, state == CheckState.Checked);
+                return;
+            }
+
+            if (Application.RenderWithVisualStyles)
```
Comment (PhyxionNL) [inline Aga.Controls/Tree/NodeControls/NodeCheckBox.cs:65]: Same.

File: Aga.Controls/Tree/NodeControls/NodePlusMinus.cs [hunk 3]
```diff
@@ -59,7 +59,14 @@ public override void Draw(TreeNodeAdv node, DrawContext context)
 				int scaledX = node.Tree.GetScaledSize(ImageSize, false);
 				int scaledY = node.Tree.GetScaledSize(ImageSize);
 				int dy = (int)Math.Round((float)(r.Height - scaledY) / 2);
-				if (Application.RenderWithVisualStyles)
+
+                if (TreeViewAdv.CustomPlusMinusRenderFunc != null)
+                {
+                    TreeViewAdv.CustomPlusMinusRenderFunc(context.Graphics, r, node.IsExpanded);
+                    return;
+                }
```
Comment (PhyxionNL) [inline Aga.Controls/Tree/NodeControls/NodePlusMinus.cs:67]: Same

File: Aga.Controls/Tree/TreeColumn.cs [hunk 4]
```diff
@@ -304,6 +310,12 @@ internal static void DrawDropMark(Graphics gr, Rectangle rect)
 
 		internal static void DrawBackground(Graphics gr, Rectangle bounds, bool pressed, bool hot)
 		{
+            if (TreeViewAdv.CustomColumnBackgroundRenderFunc != null)
+            {
+                TreeViewAdv.CustomColumnBackgroundRenderFunc(gr, bounds, pressed, hot);
+                return;
+            }
```
Comment (PhyxionNL) [inline Aga.Controls/Tree/TreeColumn.cs:317]: Formatting

File: Aga.Controls/Tree/TreeViewAdv.cs [hunk 5]
```diff
@@ -1,29 +1,36 @@
...
 	/// Features: Model/View architecture. Multiple column per node. Ability to select
-	/// multiple tree nodes. Different types of controls for each node column: 
+	/// multiple tree nodes. Different types of controls for each node column:
 	/// <see cref="CheckBox"/>, Icon, Label... Drag and Drop highlighting. Load on
 	/// demand of nodes. Incremental search of nodes.
 	/// </summary>
 	public partial class TreeViewAdv : Control
 	{
-		private const int LeftMargin = 7;
+        public static Action<Graphics, Rectangle, bool> CustomCheckRenderFunc;
```
Comment (PhyxionNL) [inline Aga.Controls/Tree/TreeViewAdv.cs:25]: Same

File: LibreHardwareMonitor/UI/Themes/Theme.cs [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Windows.Forms;
+using Aga.Controls.Tree;
+using Aga.Controls.Tree.NodeControls;
+using Microsoft.Win32;
+
+namespace LibreHardwareMonitor.UI.Themes
+{
+    public abstract class Theme
```
Comment (PhyxionNL) [inline LibreHardwareMonitor/UI/Themes/Theme.cs:13]: (Auto) format this file.

**Review by PhyxionNL — CHANGES_REQUESTED** [review]
This looks really nice 🎉 Formatting is off in almost every file though, are you using different formatting settings than default?

File: Aga.Controls/Tree/NodeControls/BaseTextControl.cs (hunk 1 above)
Comment (bjarniivarsson) [reply Aga.Controls/Tree/NodeControls/BaseTextControl.cs:222]: Yes, formatting is way off on this file (I assumed that was on purpose?). I can fix the formatting, but that's basically going to change the whole file, should I do that?

As for the other files (under LibreHardwareMonitor), I'm using VS2022 and have tried VSCode as well, both seem to be loading the .editorconfig, but I'm not getting a different formatting, what am I missing?

**Comment by JFMugen** [comment]
@bjarniivarsson 

What happened to this ;(

**Comment by TheRedfoox** [comment]
Is it still relevant ? Do you need help to take over the PR ?
We need this ♥️

File: Aga.Controls/Tree/NodeControls/BaseTextControl.cs (hunk 1 above)
Comment (MRog40) [reply Aga.Controls/Tree/NodeControls/BaseTextControl.cs:222]: I've created [#847](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/847) that has this code, with fixed merge conflicts, and another nice to have for dark theme (thicker plot lines)

**Comment by PhyxionNL** [comment]
Closing in favor of #847

**Pull request closed without merging**